src/queue/simple.rs

```diff
@@ -74,36 +73,34 @@ where
         if let Some(t) = self.local_queue.pop() {
             return Some(into_pop(t, true));
         }
-        let mut need_retry;
-        loop {
+        let mut need_retry = true;
+        while need_retry {
             need_retry = false;
             match self.injector.steal_batch_and_pop(&self.local_queue) {
                 Steal::Success(t) => return Some(into_pop(t, false)),
                 Steal::Retry => need_retry = true,
                 _ => {}
             }
-            // Steal with a random start to avoid imbalance.
-            let len = self.stealers.len();
-            if len > 0 {
-                let start_index = self.rng.gen_range(0, len);
-                for stealer in self
-                    .stealers
-                    .iter()
-                    .chain(&self.stealers)
-                    .skip(start_index)
-                    .take(len)
-                {
+            if !self.stealers.is_empty() {
```

Comment:
This line seems unnecessary

Reply:
It reduces several instructions, but I'm OK to remove it if you insist as it does not have significant performance impact in most case.